lambdas/src/apis/profiles/EnsOwnership.ts

```diff
@@ -24,87 +20,17 @@ export class EnsOwnership {
     maxSize: number,
     maxAge: number
   ) {
-    this.internalCache = new LRU({ max: maxSize, maxAge })
-  }
-
-  async areNamesOwnedByAddress(ethAddress: EthAddress, namesToCheck: Name[]): Promise<Map<Name, Owned>> {
-    const map = new Map([[ethAddress, namesToCheck]])
-    const result = await this.areNamesOwned(map)
-    return result.get(ethAddress.toLowerCase())!
-  }
-
-  async areNamesOwned(check: Map<EthAddress, Name[]>): Promise<Map<EthAddress, Map<Name, Owned>>> {
-    // Set up result
-    const result: Map<EthAddress, Map<Name, Owned>> = new Map(
-      Array.from(check.keys()).map((ethAddress) => [ethAddress.toLowerCase(), new Map()])
-    )
-
-    // We will keep the unknown names in 2 different structures, so that it is easier to use them later
-    const unknown: Name[] = []
-    const unknownMap: Map<EthAddress, Name[]> = new Map()
-
-    // Check what is on the cache
-    for (const [ethAddress, names] of check) {
-      const lowerCaseAddress = ethAddress.toLowerCase()
-      const ethAddressResult = result.get(lowerCaseAddress)!
-      for (const name of names) {
-        const owner = this.internalCache.get(name)
-        if (owner === undefined) {
-          unknown.push(name)
-          const unknownNamesPerAddress = unknownMap.get(lowerCaseAddress)
-          if (!unknownNamesPerAddress) {
-            unknownMap.set(lowerCaseAddress, [name])
-          } else {
-            unknownNamesPerAddress.push(name)
-          }
-        } else {
-          ethAddressResult.set(name, lowerCaseAddress === owner)
-        }
-      }
-    }
-
-    // Fetch owners for unknown names
-    const graphFetchResult = await this.fetchActualOwners(unknown)
-
-    // Store fetched data in the cache, and add missing information to the result
-    for (const [ethAddress, names] of unknownMap) {
-      const ethAddressResult = result.get(ethAddress)!
-      for (const name of names) {
-        const owner = graphFetchResult.get(name)
-        this.internalCache.set(name, owner ?? null) // We are setting undefined values to null. This is so that we know we queried the data, and there is no owner
-        ethAddressResult.set(name, owner === ethAddress)
-      }
-    }
-
-    return result
-  }
-
-  private async fetchActualOwners(names: Name[]): Promise<Map<Name, EthAddress>> {
-    const result: Map<Name, EthAddress> = new Map()
-    let offset = 0
-    while (offset < names.length) {
-      const namesSlice = names.slice(offset, offset + EnsOwnership.PAGE_SIZE)
-      const owners = await this.fetchOwnedNames(namesSlice)
-      for (const { name, owner } of owners) {
-        result.set(name, owner)
-      }
-      offset += EnsOwnership.PAGE_SIZE
-    }
-
-    return result
+    super(maxSize, maxAge)
   }
 
   /** This method will take a list of names and return only those that are owned by the given eth address */
-  private async fetchOwnedNames(names: Name[]) {
+  protected async querySubgraph(names: Name[]) {
```

Comment:
I don't understand the relationship between name and nft here. This function receives a list of names, but then calls those names nft. Could we keep only names or nfts? On the other hand, the old function's name was clearer in my opinion. This one is called "querySubgraph" but that doesn't give a good idea of what is it doing.

Reply:
This was changed because `querySubgraph` is inherited from `NFTOwnership`. In `EnsOwnership`, those nfts are names. I can always call them nfts if that is what you prefer, but we can't avoid the method being more "generic"